.pre-commit-config.yaml

```diff
@@ -54,6 +29,13 @@ repos:
     rev: 5.11.4
     hooks:
       - id: isort
+  - repo: https://github.com/charliermarsh/ruff-pre-commit
+    rev: v0.0.227
+    hooks:
+      - id: ruff
+        args:
+          - --fix
```

Comment:
Especially with `--fix`, I think placing ruff before black would be in order. Some of the formatting changes ruff does in certain configurations are such that black will subsequently reformat them differently than it did before, and thus the next pre-commit run would fail again.

I haven't met a ping-pong problem case between the two, but for example trailing commas added by `COM` are an example of the above. In that, a black -> ruff sequence would cause the next black run to add line breaks it did not before in spots where ruff added trailing commas. (`COM` is not enabled in the current config here, but I wouldn't be surprised if there were other cases like that.)
A tradeoff is that black formatting applied after messages from ruff could make line numbers to change, but that seems a lesser evil to me.

Haven't checked if there are similar considerations in CI configs which run hook stages selectively.

Reply:
Yeah, good call!